### PR TITLE
12.0 split beesdoo inventory

### PR DESCRIPTION
**Pull request by robinkeunen**
- [x] Split beesdo_inventory into
    1. stock_picking_responsible
    2. stock_picking_max_shipping_date
    3. stock_picking_copy_quantity
- [x] Rename xml ids (no migration scripts, the views were never inherited)
- [x] Add translations for each module
- [x] Add test coverage
- [x] + minor refactorings
- [x] submit to the OCA.

fixes :
- https://github.com/beescoop/Obeesdoo/issues/348
- https://github.com/beescoop/Obeesdoo/issues/349
- https://github.com/beescoop/Obeesdoo/issues/350

related PR :
- https://github.com/OCA/stock-logistics-workflow/pull/1026
- ~~https://github.com/coopiteasy/stock-logistics-workflow/pull/1~~
    - in this pr, we decided to remove the field max_shipping_date

Notes :
- Hereunder, we discuss wether [oca/stock_move_line_auto_fill](https://github.com/OCA/stock-logistics-workflow/tree/12.0/stock_move_line_auto_fill) can replace `stock_picking_copy_quantity`
- **RED CI** can't adapt test-requirements.txt to include `stock_picking_responsible` since pip setup files are not generated on oca/stock-logistics-workflow

### CODE REVIEW

**Comment by robinkeunen** [comment]
@carmenbianca I can't get the tests to pass because I can't adapt `test-requirements.txt` to include stock_picking_responsible since pip setup files are not generated on oca/stock-logistics-workflow . What do you suggest we do ?

cf https://github.com/OCA/stock-logistics-workflow/pull/1026

**Comment by carmenbianca** [comment]
@robinkeunen You own that PR, so you can add the setup file yourself, no?

**Comment by robinkeunen** [comment]
Ha yes, I had missed the `setup` directory on that repo. I'll try doing that.

**Comment by robinkeunen** [comment]
@victor-champonnois @carmenbianca this pr is open for reviews

@vdewulf this PR is open for tests on all databases 

`beedoo_inventory` had four functionalities : 
1. adds `responsible` field, 
	- moved to new module `stock_picking_responsible`
2. add `max_shipping_date` field 
	- removed cf https://github.com/coopiteasy/stock-logistics-workflow/pull/1
3. add button to copy quantity to stock pickings.
	- removed, test wether it can be replaced by [oca/stock_move_line_auto_fill](https://github.com/OCA/stock-logistics-workflow/tree/12.0/stock_move_line_auto_fill)
4. restricts picking's available product to those sold by supplier as "main supplier"
	- nothing to test, will be done later

**Comment by vdewulf** [comment]
@robinkeunen 

Tested on lenid-test (lenid.test.coopiteasy.be)

1. stock_picking_responsible => it's installed and there is a "responsible field" on the WH/IN (I created a PO, validated, and set the responsible from Administrator (by default) to Antonin. Antonin was added to the followers. Seems ok.


2. button to copy quantity to stock pickings. It's removed, indeed. I installed  [oca/stock_move_line_auto_fill](https://github.com/OCA/stock-logistics-workflow/tree/12.0/stock_move_line_auto_fill)
After installation, it's not possible to modify the quantity on the line of rht WH.IN directly (needs to click on the "burger" icon on the right to open a pop up where the quantity can be updated). This is annoying and we had this issue when migrating supermarket from v9 to v12. It's linked to the fact that users are set in the group "Gérer les emplacements multiples et les entrepôts"
I removed the administrator from this group and I retrieve the "normal" WH/IN view.
The button "remplissage automatique" is added my the module and works fine (the Fait column is filled in with the Demande initiale value).
CONCLUSION: ok to add the OCA module but when installation occurs on customers database, we'll have a lot of complaints if we don't remove all the users from the group "Gérer les emplacements multiples et les entrepôts". **This should be done by the scripts or functional people need to do it on each customer database after deployement.**


3. restricts picking's available product to those sold by supplier as "main supplier"
Not tested.

**Comment by robinkeunen** [comment]
@vdewulf merci pour les test

> CONCLUSION: ok to add the OCA module but when installation occurs on customers database, we'll have a lot of complaints if we don't remove all the users from the group "Gérer les emplacements multiples et les entrepôts". This should be done by the scripts or functional people need to do it on each customer database after deployement.

I added to the migration script a bit of code to remove all users from group "stock.group_tracking_lot" aka "Manage Packages" sometimes translated to "Gérer les emplacements multiples et les entrepôts".

The change was deployed on the test server on all bases who had beesdoo_inventory installed.

**Comment by vdewulf** [comment]
@robinkeunen can you tell me which DB are in this situation? I just went to Demain and Ouftcoop, they don't have beesdoo_inventory. I don't want to make it manually on all DB ;)

**Comment by robinkeunen** [comment]
@vdewulf Here you are:
```
> ociedoo list-db --module beesdoo_inventory
bablmarket-test
bees-test
coopeco-test-backup
lasource-test
lenid-test
lenid-test-polln-shift
lepedalo-test
lepedalo-test-2022-06-28
lesptitspots-test
restore-bloum-test-5o351omr
spp-test
spp-test-product-state
spp-test-theo-2022-02-28
vervicoop-test
```

**Comment by vdewulf** [comment]
@robinkeunen 
I checked on the following database and there are still users in the "stock.group_tracking_lot" group :
- spp-test
- bees (265 users in test versus 5 in prod)

On this db, the group was empty:
- babl market
I stopped the control here, something went wrong.

**Comment by robinkeunen** [comment]
@vdewulf 
Alright, digged a bit deeper. The button is visible if the user is in any of these groups :

- stock.group_tracking_lot : Manage Packages
- stock.group_stock_multi_locations : Manage Multiple Stock Locations
- stock.group_tracking_owner : Manage Different Stock Owners

I confused in stock.group_tracking_lot in my previous comment with Manage Multiple Stock Locations.

I adapted my script to remove all users from these three groups. It is being updated on all relevant databases right now.

**Comment by robinkeunen** [comment]
@vdewulf the script to remove users from groups displaying the "burger icon" was run on all supermarket database, can you test again ?

File: beesdoo_inventory/__manifest__.py [hunk 1]
```diff
@@ -11,11 +11,10 @@
     "author": "Beescoop - Cellule IT, Coop IT Easy SC",
     "website": "https://github.com/beescoop/Obeesdoo",
     "category": "Inventory",
-    "version": "12.0.1.0.1",
+    "version": "12.0.2.0.1",
```
Comment (carmenbianca) [inline beesdoo_inventory/__manifest__.py]: No need to bump. ocabot does it for us.

**Review by carmenbianca — APPROVED** [review]
one comment, otherwise LGTM.

also left a review at https://github.com/OCA/stock-logistics-workflow/pull/1026

File: beesdoo_inventory/__manifest__.py (hunk 1 above)
Comment (robinkeunen) [reply beesdoo_inventory/__manifest__.py]: I need to do it to control my migration scripts. I'll _nobump merge_ it (or _patch merge_ it ?) .

Comment (carmenbianca) [reply beesdoo_inventory/__manifest__.py]: ahh makes sense. `merge nobump` is ok

**Comment by robinkeunen** [comment]
@vdewulf avant que je ne reprenne tout à zero, si tu peux valider ceci, je le mergerai déjà.

**Comment by vdewulf** [comment]
@robinkeunen 
I checked the group "Paramètres techniques + Gérer plusieurs emplacement de stock" on the following databases:

* bablmarket-test => empty
* bees-test ==> error when goiing into the database (KeyError: 'beesdoo.shift.sheet') + I don't think it's the DB you updated in the meantime.
* lasource-test => no user ok
* lenid-test =>  no user ok
* lenid-test-polln-shift => not tested
* lepedalo-test => ok
* lepedalo-test-2022-06-28 => not tested
* lesptitspots-test => not tested
* restore-bloum-test-5o351omr => not tested
* spp-test => ok
* spp-test-product-state => not tested
* spp-test-theo-2022-02-28 => not tested
* vervicoop-test => ok

Conclusion: ok!

Just to make sure it's not forgotten, there is still this that has not been done yet:
restricts picking's available product to those sold by supplier as "main supplier"

**Comment by robinkeunen** [comment]
@vdewulf thanks! The main supplier filter depends on another refactoring. It's in this issue https://github.com/beescoop/Obeesdoo/issues/441

**Pull request merged**